src/app/core/components/directives/details-height/details-height.directive.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  Directive, ElementRef, Inject, Input, OnDestroy, OnInit,
+} from '@angular/core';
+import { WINDOW } from 'app/helpers/window.helper';
+
+@Directive({
+  selector: '[ixDetailsHeight]',
+})
+export class IxDetailsHeightDirective implements OnInit, OnDestroy {
+  @Input() ixDetailsHeightParantClass: string;
+  @Input() hiddenHeight = 230;
+  @Input() maxShowHeight = 150;
+
+  readonly onScrollHandler = this.onScroll.bind(this);
+  heightDetails = `calc(100vh - ${this.hiddenHeight}px)`;
+
+  constructor(
+    @Inject(WINDOW) private window: Window,
+    private el: ElementRef,
+  ) {}
+
+  ngOnInit(): void {
+    this.el.nativeElement.style.height = this.heightDetails;
+    this.window.addEventListener('scroll', this.onScrollHandler, true);
```

Comment:
MDN recommends throttling scroll listeners. Should we?

Reply:
I suppose throttling of this scroll listener would be an overkill, and would potentially degrade scrolling animation, by making it sluggish